spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/channel/MessageGroupPreparedStatementSetter.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2002-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jdbc.store.channel;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Types;
+
+import org.springframework.core.serializer.support.SerializingConverter;
+import org.springframework.integration.IntegrationMessageHeaderAccessor;
+import org.springframework.integration.util.UUIDConverter;
+import org.springframework.jdbc.support.lob.LobHandler;
+import org.springframework.messaging.Message;
+
+/**
+ * Callback to be used with {@link org.springframework.integration.jdbc.store.JdbcChannelMessageStore}.
+ * <p>
+ * Behavior is same as standard {@link org.springframework.jdbc.core.PreparedStatementSetter}, it takes in
+ * extra {@code Message<?> requestMessage}, {@code Object groupId},
+ * {@code String region} and {@code boolean priorityEnabled} as parameters used
+ * for {@code addMessageToGroup} method in {@link org.springframework.integration.jdbc.store.JdbcChannelMessageStore}.
+ *
+ * @author Meherzad Lahewala
+ * @since 5.0
+ * @see org.springframework.jdbc.core.PreparedStatementSetter
+ */
+public class MessageGroupPreparedStatementSetter {
```

Comment:
Well, after closer look, it really isn't a `group`. It just does nothing with groups. This is exactly about insertion a single message.
Will rename to the `MessagePreparedStatementSetter` during merge...

Reply:
Hm. We have already one - `org.springframework.integration.jdbc.MessagePreparedStatementSetter`.
so, no choice: this one will be `ChannelMessageStorePreparedStatementSetter` - to really distinguish this one for target purpose and don't let end-user to be confused.